Deduplicate the correctionIndicatorType method.

nmahendru opened the pull request:
#### c213889156180d79d3d01c5108e2152c2b318c40
<pre>
Deduplicate the correctionIndicatorType method.
<a href="https://bugs.webkit.org/show_bug.cgi?id=248216">https://bugs.webkit.org/show_bug.cgi?id=248216</a>
<a href="https://rdar.apple.com/102760220">rdar://102760220</a>

Reviewed by Alex Christensen.

Move the correctionIndicatorType into WebCore and use that
instead in WebKit/WebKitLegacy. Earlier, they both had the exact same function individually.

* Source/WebCore/page/mac/CorrectionIndicator.h: Added.
* Source/WebCore/page/mac/CorrectionIndicator.mm: Added.
(WebCore::correctionIndicatorType):

* Source/WebCore/SourcesCocoa.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/page/mac/CorrectionIndicator.h:
* Source/WebCore/page/mac/CorrectionIndicator.mm:
(WebCore::correctionIndicatorType):
* Source/WebKit/UIProcess/mac/CorrectionPanel.mm:
(correctionIndicatorType): Deleted.
* Source/WebKitLegacy/mac/WebCoreSupport/CorrectionPanel.mm:
(correctionIndicatorType): Deleted.
* Source/WebCore/page/ios/FrameIOS.mm:

Canonical link: <a href="https://commits.webkit.org/272388@main">https://commits.webkit.org/272388@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2d30facbaefb286224b2e33d128a3bb79eb2cf71

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/31489 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/10164 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/33193 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/33977 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/28519 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/32259 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/12512 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/7411 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/28133 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/31829 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/8555 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/28100 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/7362 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/7523 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/28009 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/35321 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/28617 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/28455 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/33656 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/7603 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/5621 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/31499 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/9260 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7394 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/8292 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/8108 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebCore/dom/Document.cpp:
```diff
@@ -27,7 +27,7 @@
 #include "config.h"
 #include "Document.h"
-
+#include "DocumentInlines.h"
```
This looks likely incorrect to add in this PR, but I'm not sure.

nmahendru replied:
Yea I was trying a few things last night. This PR was draft so I was planning to clean it up after I got a successful build first. But thanks for reviewing it anyway!

achristensen07 commented on Source/WebCore/page/mac/CorrectionIndicator.h:
```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#if PLATFORM(MAC)
+#include "AlternativeTextClient.h"
```
This can probably be replaced by just a forward declaration, which results in less build time increase:
enum class AlternativeTextType : uint8_t;

nmahendru replied:
thanks. Done now!

achristensen07 commented on Source/WebCore/page/mac/CorrectionIndicator.mm:
```diff
@@ -0,0 +1,48 @@
+
+#if PLATFORM(MAC)
+
+#include <WebCore/CorrectionIndicator.h>
```
This should be #include "CorrectionIndicator.h" and it should probably be the line right after config.h

nmahendru replied:
I was just trying to keep it inside the `#if PLATFORM(MAC)` as it will not even be there where it's not needed ?

achristensen07 commented on Source/WebCore/page/mac/CorrectionIndicator.h:
```diff
@@ -0,0 +1,33 @@
+#if PLATFORM(MAC)
+#include "AlternativeTextClient.h"
+namespace WebCore {
+WEBCORE_EXPORT NSCorrectionIndicatorType correctionIndicatorType(WebCore::AlternativeTextType);
```
empty line before and after.

nmahendru replied:
done.

achristensen07 commented on Source/WebCore/page/mac/CorrectionIndicator.mm:
```diff
@@ -0,0 +1,48 @@
+#if PLATFORM(MAC)
+
+#include <WebCore/CorrectionIndicator.h>
+namespace WebCore {
```
newline after

nmahendru replied:
done.

achristensen07 commented on Source/WebCore/page/mac/CorrectionIndicator.mm:
```diff
@@ -0,0 +1,48 @@
+        return NSCorrectionIndicatorTypeDefault;
+    }
+}
+}
```
newline before and probably after

nmahendru replied:
done.

achristensen07 commented on Source/WebKit/UIProcess/mac/CorrectionPanel.mm:
```diff
@@ -28,27 +28,12 @@
 
 #if USE(AUTOCORRECTION_PANEL)
 
+#import <WebCore/CorrectionIndicator.h>
```
imports with <> go after imports with "", each in lexical order

nmahendru replied:
done.

achristensen07 commented on Source/WebCore/SourcesCocoa.txt:
```diff
@@ -220,6 +220,7 @@ loader/mac/DocumentLoaderMac.cpp
 loader/mac/LoaderNSURLExtras.mm
 loader/mac/ResourceLoaderMac.mm
 page/CaptionUserPreferencesMediaAF.cpp
+page/mac/CorrectionIndicator.mm
```
This should be with the other files in page/Mac, sorted lexically.

nmahendru replied:
good catch. done!

achristensen07 commented on Source/WebCore/WebCore.xcodeproj/project.pbxproj:
```diff
@@ -3646,6 +3646,7 @@
 		9479493F1E0463C600018D85 /* DeprecatedCSSOMRect.h in Headers */ = {isa = PBXBuildFile; fileRef = 947949281E032AE900018D85 /* DeprecatedCSSOMRect.h */; settings = {ATTRIBUTES = (Private, ); }; };
 		947949401E0463CE00018D85 /* DeprecatedCSSOMPrimitiveValue.h in Headers */ = {isa = PBXBuildFile; fileRef = 947949261E03132000018D85 /* DeprecatedCSSOMPrimitiveValue.h */; settings = {ATTRIBUTES = (Private, ); }; };
 		947949411E0463D400018D85 /* DeprecatedCSSOMCounter.h in Headers */ = {isa = PBXBuildFile; fileRef = 9479492A1E032EDB00018D85 /* DeprecatedCSSOMCounter.h */; settings = {ATTRIBUTES = (Private, ); }; };
+		94925CE92B320BCC001C62B8 /* CorrectionIndicator.h in Headers */ = {isa = PBXBuildFile; fileRef = 94925CE62B320BCC001C62B8 /* CorrectionIndicator.h */; settings = {ATTRIBUTES = (Private, ); }; };
```
It looks like you need to run "Tools/Scripts/sort-Xcode-project-file Source/WebCore/WebCore.xcodeproj"

nmahendru replied:
as discussed, running it did nothing to this file. So leaving it as is.

achristensen07 commented on Source/WebCore/dom/Document.cpp:
```diff
@@ -27,7 +27,6 @@
 
 #include "config.h"
 #include "Document.h"
-
```
This change should be undone.  WebKit style has config, the header with the same name as the cpp/mm, then a newline then other headers.

nmahendru replied:
ahh. Undid the change.

achristensen07 commented on Source/WebCore/page/mac/CorrectionIndicator.mm:
```diff
@@ -0,0 +1,53 @@
+#include "config.h"
+
+#if PLATFORM(MAC)
+#include "CorrectionIndicator.h"
```
Let's put this immediately after config.h, before if PLATFORM(MAC)

nmahendru replied:
done

nmahendru committed:
Deduplicate the correctionIndicatorType method.
https://bugs.webkit.org/show_bug.cgi?id=248216
rdar://102760220

Reviewed by Alex Christensen.

Move the correctionIndicatorType into WebCore and use that
instead in WebKit/WebKitLegacy. Earlier, they both had the exact same function individually.

* Source/WebCore/page/mac/CorrectionIndicator.h: Added.
* Source/WebCore/page/mac/CorrectionIndicator.mm: Added.
(WebCore::correctionIndicatorType):

* Source/WebCore/SourcesCocoa.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/page/mac/CorrectionIndicator.h:
* Source/WebCore/page/mac/CorrectionIndicator.mm:
(WebCore::correctionIndicatorType):
* Source/WebKit/UIProcess/mac/CorrectionPanel.mm:
(correctionIndicatorType): Deleted.
* Source/WebKitLegacy/mac/WebCoreSupport/CorrectionPanel.mm:
(correctionIndicatorType): Deleted.
* Source/WebCore/page/ios/FrameIOS.mm:

Canonical link: https://commits.webkit.org/272388@main
